### PR TITLE
Add USB and nRF52840 dongle support.

### PR DESCRIPTION
**Pull request by e-rk**
Replace UART transport with USB CDC ACM.
Add support for nRF52840 dongle.
Add DFU trigger capability allowing the application to enter DFU mode
without pressing the reset button.
Increase internal buffer size for captured packets.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,15 +59,22 @@ Ensure that Python directory is included in your `PATH` system environment varia
...
 5. Click 'Start'.
 
-## Custom Wireshark dissector
+## Wireshark configuration
+
+### Capturing Thread packets
+
+In order to capture packets exchanged on the Thread network it is necessary to configure the Wireshark to use correct decryption keys.
+To set decryption keys go to `Edit -> Preferences... -> Protocols -> IEEE 802.15.4 -> Decryption Keys`. The default decryption key used by nRF5 SDK for Thread and Zigbee examples is `00112233445566778899aabbccddeeff`. Set decryption key index to `0` and `Key hash` to `Thread hash` value.
+
```
Comment (LuDuda) [inline README.md]: I think more steps are required.

1. Set up Master Key (you have that one)
2. Thread CoAP - should be set to 61631 i believe (need to verify)
3.  Set up 6LoWPAN contextes (Context 0: fdde:ad00:beef:0::/64, Context 1: fd11:22::/64)
4. Disable ZigBee NWK protocol (and any other unwanted protocols):
Click Analyze and then select Enabled Protocols and Uncheck the following: 
- LwMesh 
- ZigBee
- ZigBee Green Power

File: README.md [hunk 2]
```diff
@@ -18,11 +18,28 @@ The software provided has been tested with the nRF52840-DK board and the followi
...
 2. Flash the firmware with the following command:
 ```
 nrfjprog -f nrf52 --program nrf802154_sniffer/nrf802154_sniffer.hex --chiperase -r
 ```
+3. J2 USB port can now be optionally disconnected.
+4. Connect the nRF52840-DK to J3 nRF USB port.
+
+__Note:__ Sniffer firmware can no longer transport captured packets over J2 USB port. Capture can be carried out using only the J3 nRF USB port.
+
+#### nRF52840-Dongle
```
Comment (LuDuda) [inline README.md]: nRF52840 Dongle (PCA10059)

File: README.md [hunk 3]
```diff
@@ -18,11 +18,28 @@ The software provided has been tested with the nRF52840-DK board and the followi
 To start using the sniffer, you must flash the firmware, install the script, and configure the sniffer in Wireshark.
 
 ### Flash firmware
+
+#### nRF52840-DK
```
Comment (LuDuda) [inline README.md]: nRF52840 DK (PCA10056)

File: README.md (hunk 1 above)
Comment (LuDuda) [reply README.md]: It's required by Thread Certification. Certain messages are incorrectly interpreted as LwMesh or Zigbee Green Power :)

File: README.md [hunk 4]
```diff
@@ -59,22 +59,42 @@ Ensure that Python directory is included in your `PATH` system environment varia
...
+* apply it on per-capture basis by adding entry to `Analyze -> Decode as...`. Set `Field`
+column to `UDP port`, `Value` to `61631` and decode using CoAP dissector (`Current` column).
+
+### 6loWPAN contexts
+
+Go to 6loWPAN settings in `Preferences` window and add values:
+* Context 0: `fdde:ad00:beef:0::/64`
+* Context 1: `fd11:22::/64`
+
+### (Optional) Disable unwanted protocols
```
Comment (LuDuda) [inline README.md]: nit: Maybe let's name it: Disable unwanted protocols (optional)

File: README.md [hunk 5]
```diff
@@ -59,22 +59,42 @@ Ensure that Python directory is included in your `PATH` system environment varia
...
+* apply the setting globally by editing CoAP protocol settings in `Preferences` and changing the `CoAP UDP port` value to `61631`.
+* apply it on per-capture basis by adding entry to `Analyze -> Decode as...`. Set `Field`
+column to `UDP port`, `Value` to `61631` and decode using CoAP dissector (`Current` column).
+
+### 6loWPAN contexts
+
+Go to 6loWPAN settings in `Preferences` window and add values:
+* Context 0: `fdde:ad00:beef:0::/64`
+* Context 1: `fd11:22::/64`
+
```
Comment (LuDuda) [inline README.md]: Add note that those contextes may vary depending of the Thread Network Data. Those values are default for our SDK.

File: README.md (hunk 1 above)
Comment (stig-bjorlykke) [reply README.md]: We should probably have a look at the heuristics for LwMesh and ZigBee regarding this packets. Please send me a capture file with incorrectly dissected packets and I'll have a look.

Comment (LuDuda) [reply README.md]: @stig-bjorlykke Thank you for offering help. If i will find time i will send you some captures.

**Pull request merged**